### ISSUE TITLE
Suddenly having issues converting MediaPicker(s)

### ISSUE THREAD

**Issue by markadrake** [Jumoo/uSyncMigrations#152]
Hello Kevin & Umbraco Community,

The (obsolete) property editor `Umbraco.MediaPicker` is causing me pain in a recent migration. While I see the correct GUID across all config files, the **conversion step** doesn't translate the GUID into a `{ key, mediaKey }` pair.

I've confirmed that the media's GUID is correct. The last entry below is produced by uSync after I manually edit the content in the backoffice, choose the correct media file, and save and publish the content node.

<Details>
<Summary>Environment Details</Summary>
<br>

Software | Umbraco 7 | Umbraco 10
---|---:|---:
Umbraco|7.15.10|10.5.1
uSync|4.0.16|10.6.1
uSync.Content|4.1.9.1|- - -
uSync.Core|6.0.15|- - -
uSync.Migrations|- - -|4.0.0-phase.3

</Details>

1. How should media GUIDs look (1) after export and (2) after conversion?
2. How would you recommend I continue debugging or troubleshooting this issue?
3. Do you suggest running any other packages or tools in the Umbraco 7 environment, prior to running a full export?

## Export from uSync in Umbraco 7:

_This is from the file produced by uSync's "full export"._

```
<featuredArticleImage>13d3c008-0e9c-4916-a99b-a4a84c371c84</featuredArticleImage>
```

## Converted by uSync in Umbraco 10:

_This is from the file produced by uSync.Migrations' "convert files"._

```
<featuredArticleImage>
  <Value><![CDATA[13d3c008-0e9c-4916-a99b-a4a84c371c84]]></Value>
</featuredArticleImage>
```

## Edited and Published in Umbraco 10:

_This is from the file produced by uSync after editing the content through Umbraco, assigning the correct image, and saving & publishing._

```
<featuredArticleImage>
  <Value><![CDATA[[
{
  "key": "3adcbc38-c33d-471e-bc52-f602844c6d8b",
  "mediaKey": "13d3c008-0e9c-4916-a99b-a4a84c371c84"
}
]]]></Value>
  </featuredArticleImage>
```

<br>

> As always - the time, effort, and products you share with the community make all of our work easier. My clients benefit so much from your efforts because I'm able to spend more time elsewhere. 
> Just want to say thanks for all your help,
> – Mark

**Comment by markadrake** [comment]
Closing this issue. The inconsistency with my current migration(s) was due to (my) user error.

**Comment by mistyn8** [comment]
I'm also having some issues.. maybe I'm making the same user error.. care to elaborate on your fix/realisation?

**Comment by markadrake** [comment]
> I'm also having some issues.. maybe I'm making the same user error.. care to elaborate on your fix/realisation?

This was so long ago all I can do is offer an apology for not going into more detail. I don't exactly remember the issue. I will share quite a silly story of a migration similar to this that caused me a lot of trouble.

This site had over 20,000 content nodes and even more media nodes (48k if I remember).

This was an Umbraco 7 migration with all your normal suspects: deprecated OOTB fields, third party packages no longer supported, first party code that needed to be refactored, etc.

A full export/import took a very long time. So I had the bright idea while troubleshooting the import (and my custom logic for importing) that I'd delete the uSync media folder so I didn't have to wait an additional 15-20 mins for each run.

I think I discovered during this project that if the media config files aren't in the uSync folder, then the data in my content nodes would be dropped or incorrect. Which makes sense

TLDR;
I foolishly deleted 'uSync/migrations/{name}/media' folder in an effort to save time. I think this may be related to the issue I opened above.

Sorry I didn't give more details when I was experiencing this problem and I'll try to do better in future tickets/comments!
